internal/controller/isbservicerollout_controller.go

```diff
@@ -140,31 +147,38 @@ func (r *ISBServiceRolloutReconciler) Reconcile(ctx context.Context, req ctrl.Re
 
 	numaLogger.Debug("reconciliation successful")
 
-	return ctrl.Result{}, nil
+	return result, nil
 }
 
 // reconcile does the real logic
-func (r *ISBServiceRolloutReconciler) reconcile(ctx context.Context, isbServiceRollout *apiv1.ISBServiceRollout) error {
+func (r *ISBServiceRolloutReconciler) reconcile(ctx context.Context, isbServiceRollout *apiv1.ISBServiceRollout) (ctrl.Result, error) {
 	numaLogger := logger.FromContext(ctx)
 
 	// is isbServiceRollout being deleted? need to remove the finalizer so it can
 	// (OwnerReference will delete the underlying ISBService through Cascading deletion)
 	if !isbServiceRollout.DeletionTimestamp.IsZero() {
 		numaLogger.Info("Deleting ISBServiceRollout")
 		if controllerutil.ContainsFinalizer(isbServiceRollout, finalizerName) {
+			GetPauseModule().deleteISBServicePauseRequest(isbServiceRollout.Namespace, isbServiceRollout.Name)
 			controllerutil.RemoveFinalizer(isbServiceRollout, finalizerName)
 		}
 		// generate metrics for ISB Service deletion.
 		r.customMetrics.DecISBServiceMetrics(isbServiceRollout.Name, isbServiceRollout.Namespace)
-		return nil
+		return ctrl.Result{}, nil
 	}
 
 	// add Finalizer so we can ensure that we take appropriate action when CRD is deleted
 	if !controllerutil.ContainsFinalizer(isbServiceRollout, finalizerName) {
 		controllerutil.AddFinalizer(isbServiceRollout, finalizerName)
 	}
 
-	obj := kubernetes.GenericObject{
+	_, pauseRequestExists := GetPauseModule().getISBSvcPauseRequest(isbServiceRollout.Namespace, isbServiceRollout.Name)
+	if !pauseRequestExists {
+		// this is just creating an entry in the map if it doesn't already exist
+		GetPauseModule().newISBServicePauseRequest(isbServiceRollout.Namespace, isbServiceRollout.Name)
```

Comment:
map entry for this ISBService gets added when the ISBService is added, and gets deleted when ISBService is deleted. We always persist memory to indicate the pause state that ISBService requires; otherwise, the PipelineRollout reconciler has incomplete information.